### DISCUSSION TITLE
Support Shiki's `defaultColor` option in Astro's `<Code/>` component

### DISCUSSION THREAD

**Discussion by ARipeAppleByYoursTruly** [withastro/roadmap#897] [Proposal]
### Body

# Summary

Make Astro's `<Code/>` component support Shiki's `defaultColor` option for better control on dual/multiple themes styling.

# Background & Motivation

By default, Shiki uses CSS' `!important` flag to override the default theme's styles to another. This approach introduces a problem when the user wants to "reverse" the default theme, ie. apply dark theme as default, as `!important` ignores cascading.

In order to solve this problem, or avoid using `!important` in general, the `defaultColor` option should be set explicitly, as Shiki defaults `defaultColor` to `"light"`, detailed in [Shiki docs](https://shiki.style/guide/dual-themes#without-default-color). But Astro's `<Code/>` component doesn't expose `defaultColor`, and so here we are.

I'm aware that [Expressive Code](https://expressive-code.com/) has solved this problem, but it would be nice to give the user another option to work with.

# Goals

- Add `defaultColor` prop to Astro's `<Code/>` component, its value will be passed to `highlighter.codeToHtml()`
- Make `defaultColor` behave like Shiki's for consistency

# Example

I've tried implementing it and seems successful in [this repo](https://github.com/ARipeAppleByYoursTruly/astro-code-defaultColor-support-demo). Changes I've made are in the `node_modules - modified files` directory.

**Comment by ARipeAppleByYoursTruly** [comment]
Bumping for visibility as this got stale

**Reply by ematipico** [reply]
This feature is small. If you want, try to send a PR. :)

To me, it seems a reasonable request

**Comment by bluwy** [comment]
There's also an issue opened in Astro respectively https://github.com/withastro/astro/issues/11238, but that's for the `shikiConfig`, not `<Code />`. However, if we do support the option, we can (and probably should) support in both places at once.

Definitely welcome a PR though.

**Comment by ARipeAppleByYoursTruly** [comment]
I'll try making a PR after withastro/astro#11238's PR have been merged, thanks for the green light

**Reply by bluwy** [reply]
As a heads up, we usually merge feature PRs for minors every 2 weeks on Wednesday (the next is Wednesday next week). So if you'd like to have the feature in soon, it may be good to create a PR based on https://github.com/withastro/astro/issues/11238 so it's also ready during then, or we'd have to wait another 2 weeks if the PR comes later.

**Reply by ematipico** [reply]
@bluwy I think we will skip next week. There's a thread on discord to talk about it
